ans2ask-ui/src/components/QuestionGrid/QuestionGrid.jsx

```diff
@@ -39,34 +39,28 @@ export default function QuestionGrid({searchQuery, selectedOption, selectedSubje
     fetchCourses();
   }, [selectedSubject]);
 
-  let content;
-  if(searchQuery.length === noQuery){
-    if(selectedOption === question){
-      if(selectedSubject === allSubjects){  
-        content = questions;
-      }
-      else {
-        content = questions.filter(question => question.subject === selectedSubject);
-      }
-    }
-    else if(selectedOption === course){
-      content = courses.items;
+  function getContent() {
+    if(searchQuery.length !== noQuery){
+      let currentContent = questions.filter(question => {
+        const titleMatches = question.title.toLowerCase().includes(searchQuery.toLowerCase());
+        const textMatches = question.body.toLowerCase().includes(searchQuery.toLowerCase());
+        return titleMatches || textMatches;
+      });
+      return currentContent; 
     }
+    if(selectedOption === course) return courses.items;
+    if(selectedSubject !== allSubjects) return questions.filter(question => question.subject === selectedSubject);
+    return questions;
   }
```

Comment:
This is much more readable! thanks miguel!

Reply:
its improving thanks to the feedback!!!